name-emu/src/main.rs

```diff
@@ -80,30 +85,43 @@ fn main() -> DynResult<()> {
 
   let args_strings: Vec<String> = env::args().collect();
 
-  if args_strings.len() > 2 {
-      return Err("USAGE: name-emu [optional port number]".into());
+  if args_strings.len() != 4 {
+      return Err("USAGE: name-emu [port number] [source file] [object file]".into());
   }
   let log_path = std::path::Path::join(env::temp_dir().as_path(), "name_log.txt");
   let mut file = File::create(log_path)?;
   file.write_all(b"NAME Development Log\n")?;
 
 
-  let (in_port, out_port): (Box<dyn std::io::Read>, Box<dyn std::io::Write>) = if let Some(port_string) = args_strings.get(1) {
-    if let Ok(port_number) = port_string.parse::<u32>() {
+  let port_string = args_strings.get(1).unwrap();
+  
+  let (in_port, out_port) = if let Ok(port_number) = port_string.parse::<u32>() {
 
-      let listener = TcpListener::bind(format!("127.0.0.1:{}", port_number)).unwrap();
+      if let Ok(listener) = TcpListener::bind(format!("127.0.0.1:{}", port_number)) {
 
-      let (stream, _) = listener.accept().unwrap();
+        let (stream, _) = listener.accept().unwrap();
 
-      // let stream = std::rc::Rc::new(stream);
-      (Box::new(stream.try_clone().unwrap()), Box::new(stream))
-    }
-    else {
-      (Box::new(std::io::stdin()), Box::new(std::io::stdout()))
-    }
+        // let stream = std::rc::Rc::new(stream);
```

Comment:
General nitpick: please remove commented code unless it's going to come back eventually maybe